app/src/main/java/covid/trace/morocco/PreferencesExtensions.kt

```diff
@@ -0,0 +1,28 @@
+package covid.trace.morocco
+
+import android.content.SharedPreferences
+import androidx.core.content.edit
+
+inline fun <reified T> SharedPreferences.get(key: String?, defaultValue: T): T {
+    when (T::class) {
+        Boolean::class -> return this.getBoolean(key, defaultValue as Boolean) as T
+        Float::class -> return this.getFloat(key, defaultValue as Float) as T
+        Int::class -> return this.getInt(key, defaultValue as Int) as T
+        Long::class -> return this.getLong(key, defaultValue as Long) as T
+        String::class -> return this.getString(key, defaultValue as String) as T
+    }
+
+    return defaultValue
+}
+
+inline fun <reified T> SharedPreferences.put(key: String?, value: T) {
```

Comment:
`key` can be non optional according to Android docs.

Reply:
Resolved